### PR TITLE
PPTP-2016 | make form take amount as maximum

### PR DESCRIPTION
**Pull request by liampan**
### Description of Work carried through

Brief description of what/why/how.

#### Check list 
 - [ ] `./precheck` was executed (Integration/Component/Unit tests)
 - [ ] `sbt scalafmt test:scalafmt` was executed 
 - [ ] Required Environment Config has been amended/added
 - [ ] Links to dependencies have been included (BE/FE work)
 - [ ] User Acceptance Tests (UAT) were run locally and have passed.
 - [ ] No Accessibility errors/warnings reported by Wave

### CODE REVIEW

File: test/forms/ConvertedPackagingCreditFormProviderSpec.scala [hunk 1]
```diff
@@ -68,5 +69,12 @@ class ConvertedPackagingCreditFormProviderSpec extends DecimalFieldBehaviours {
                                    FormError(fieldName, "convertedPackagingCredit.error.required")
       )
     }
+
+    "must reject greater than the allowed amount" in {
```
Comment (liampan) [inline test/forms/ConvertedPackagingCreditFormProviderSpec.scala:73]: this is probably a duplicate test of the scaffold ones, however as this is specific behaviour i think it makes it explicit to have this test

File: app/controllers/ConvertedPackagingCreditController.scala [hunk 2]
```diff
@@ -62,7 +63,9 @@ class ConvertedPackagingCreditController @Inject() (
       implicit request =>
         val pptId: String = request.request.enrolmentId.getOrElse(throw new IllegalStateException("no enrolmentId, all users at this point should have one"))
 
-        form.bindFromRequest().fold(
+        val userCredit = 100 //todo this amount will come from PPTP-2015
```
Comment (PSouthern) [inline app/controllers/ConvertedPackagingCreditController.scala:66]: Will this max value break the Acceptance Tests?

**Review by PSouthern — COMMENTED** [review]
I think the ATs will need to be updated as a result of the max value.

File: app/forms/ConvertedPackagingCreditFormProvider.scala [hunk 3]
```diff
@@ -20,17 +20,16 @@ import forms.mappings.Mappings
...
+class ConvertedPackagingCreditFormProvider extends Mappings {
 
-  def apply(): Form[BigDecimal] =
+  def apply(maximumAllowed: BigDecimal): Form[BigDecimal] =
     Form(
       "value" -> currency("convertedPackagingCredit.error.required",
                           "convertedPackagingCredit.error.wholeNumber",
                           "convertedPackagingCredit.error.nonNumeric"
+      ).verifying(
+        inRange[BigDecimal](0.01, maximumAllowed, "convertedPackagingCredit.error.outOfRange")
```
[the remark this answers is not in the document]
Comment (liampan) [reply app/forms/ConvertedPackagingCreditFormProvider.scala]: 0 is now allowed, this was a previously missed requirement. fixed in this pr

**Pull request merged**